### ISSUE TITLE
Editable Installer: Handle changes cleanly

### ISSUE THREAD

**Issue by abn** [python-poetry/poetry#2855]
Today, the editable installer (`poetry>1.0.10`), does not handle removals of files (includes, console scripts etc.) cleanly. This needs to be improved. 

There are two potential approaches,
1. Handle the delta manually by using the `*.dist-info/RECORD` csv file. We could load the list prior to new install and then identify the delta after installation.
2. Uninstall editable package explicitly (this today might need to use pip until poetry has an internal mechanism) prior to installation again. This might be cleaner as this should separate the concerns cleanly.

### Reproduction
This issue can be reproduced as follows.
```sh
poetry new foo
pushd foo
sed -i /pytest/d pyproject.toml
echo 'def main(): print("hello world")' >> foo/__init__.py
printf '\n\n[tool.poetry.scripts]\nhello = "foo:main"' >> pyproject.toml
poetry install
sed -i s/hello/world/g pyproject.toml
poetry install
```
Today, once this is done, the following happens.
```console
$ poetry run type hello
hello is /tmp/foo/.venv/bin/hello
```

We expect this to happen.
```console
$ poetry run type hello
/usr/bin/type: line 2: type: hello: not found
```

**Comment by bhallaY** [comment]
Hi, I would like to attempt to resolve this issue! I can try to follow way 2 you specify.

**Comment by finswimmer** [comment]
Hello @bhallaY ,

do you still like to work on this issue?

fin swimmer

**Comment by akeeman** [comment]
Just to be complete, as https://github.com/python-poetry/poetry/issues/3289 mentions, a set of related reproduction steps:

```bash
# assuming poetry 1.1.4, in-project venv, and python 3.8:
poetry new foo
cd foo
poetry install
poetry version minor
poetry install
```
then 
```bash
ls .venv/lib/python3.8/site-packages/foo*.dist-info
```
outputs something similar to
```
.venv/lib/python3.8/site-packages/foo-0.1.0.dist-info:
INSTALLER  METADATA  RECORD

.venv/lib/python3.8/site-packages/foo-0.2.0.dist-info:
INSTALLER  METADATA  RECORD
```
Therefore
```
poetry run python -c "from importlib.metadata import version; print(version('foo'))"
```
outputs
```
0.1.0
```
instead of the expected
```
0.2.0
```

**Comment by abn** [comment]
Resolved by: #3900
